packages/opencensus-web-instrumentation-zone/src/zone-types.ts

```diff
@@ -67,3 +67,24 @@ export declare interface WindowWithOcwGlobals extends Window {
   // That way the header is not added to all xhrs.
   ocTraceHeaderHostRegex?: string | RegExp;
 }
+
+/**
+ * Allows to keep track of performance entries related to a XHR.
+ * As some XHRs might generate a CORS pre-flight request, the XHR
+ * might have either two performance resource entries or a single
+ * performance resource entry.
+ */
+export type XhrPerformanceResourceTiming =
```

Comment:
Rather than making this either a tuple or timing, what would you think about making this always an array?

My thinking is that if it's always an array then some of the types and code could be easier to understand. It also unifies the way you compare start/end time in the pair case and the single request case (both cases use the start time of the first element and the end time of the last element)

Reply:
Another option might be an object with fields like
{
  preFlightRequest?: PerformanceResourceTiming,  // optional
  mainRequest: PerformanceResourceTiming,
}

That might make it more explicit to the reader what each piece means, vs using an array index. Each approach will have its pros and cons so up to you what seems best!

Reply:
That's a good idea too - I didn't think of an object based approach!

Reply:
I decided for the object idea, as I think makes the code easier to read and better to explain the algorithm steps and tests.